File: backend/Models/Task.cs
using System.ComponentModel.DataAnnotations;

namespace TaskManagementApi.Models;

public class TaskItem
{
    public required int Id { get; set; } 
    public required string Title { get; set; }
    public required string Description { get; set; }
    public Status Status { get; set; } = Status.Pending;
    public DateTime? DueDate { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? LastUpdated { get; set; } = DateTime.UtcNow;
    [Required]
    public string CreatorId { get; set; } = null!;
    // public ICollection<TaskAssignment>? TaskAssignments { get; set; }
    public Priority? Priority { get; set; }
    public string? Category { get; set; }
    public List<string>? Tags { get; set; }
}

public enum Status
{
    Pending,
    In_Progress,
    Completed,
    On_Hold,
    Cancelled
}

public enum Priority
{
    Low,
    Medium,
    High,
    Urgent
}

File: front/src/types/task.ts
export enum Status {
  Pending = "PENDING",
  InProgress = "IN_PROGRESS",
  Completed = "COMPLETED",
  OnHold = "ON_HOLD",
  Cancelled = "CANCELLED",
}

export enum Priority {
  Low = "LOW",
  Medium = "MEDIUM",
  High = "HIGH",
  Urgent = "URGENT",
}

export interface Task {
  id: number;
  title: string;
  description: string;
  status: string;
  dueDate?: string;
  creatorId: string;
  priority: string | null;
  category?: string;
  tags?: string[];
  createdAt: string;
  lastUpdated: string;
}

export interface TaskFormData {
  title: string;
  description: string;
  status: Status | string;
  priority: Priority | string | null;
  category?: string;
  tags: string[];
  dueDate?: string;
}

export interface EditState {
  isEditing: boolean;
  taskId: number | null;
  form: TaskFormData;
}
